src/app/page.tsx

```diff
@@ -1,14 +1,74 @@
 import { DndWrapper } from '@/hoc/DndWrapper'
 import { TaskBoard } from './ui/TaskBoard'
 import { Header } from '@/components/layouts/Header'
+import { WorkflowStateResponse } from '@/types/dto/workflowStates.dto'
+import { apiUrl } from '@/config'
+import { ClientSideStateUpdate } from '@/hoc/ClientSideStateUpdate'
+import { TaskResponse } from '@/types/dto/tasks.dto'
+import { IAssignee } from '@/types/interfaces'
+import { addTypeToAssignee } from '@/utils/addTypeToAssignee'
+import { handleCreate, updateWorkflowStateIdOfTask } from './actions'
+
+export const revalidate = 0
+
+async function getAllWorkflowStates(token: string): Promise<WorkflowStateResponse[]> {
+  const res = await fetch(`${apiUrl}/api/workflow-states?token=${token}`, {
+    next: { tags: ['getAllWorkflowStates'] },
+  })
+
+  const data = await res.json()
+
+  return data.workflowStates
+}
+
+async function getAllTasks(token: string): Promise<TaskResponse[]> {
+  const res = await fetch(`${apiUrl}/api/tasks?token=${token}`, {
+    next: { tags: ['getAllTasks'], revalidate: 0 },
+  })
+
+  const data = await res.json()
+
+  return data.tasks
+}
+
+async function getAssigneeList(token: string): Promise<IAssignee> {
+  const res = await fetch(`${apiUrl}/api/users?token=${token}`, {
+    next: { tags: ['getAssigneeList'], revalidate: 0 },
+  })
+
+  const data = await res.json()
+
+  return data.users
+}
+
+export default async function Main({ searchParams }: { searchParams: { token: string } }) {
+  const token = searchParams.token
+
+  if (!token) {
+    throw new Error('Please pass the token!')
+  }
+
+  const workflowStates = await getAllWorkflowStates(token)
+  const tasks = await getAllTasks(token)
```

Comment:
We haven't created the tasks for real-time updates already. This feature will come in the next sprint. 

The approach is as follows:
- For the first render, tasks will be fetched in the server side so that we can also leverage SSR on the first render. This is also important because we are using `revalidation` i.e. `revalidateTag('getAllTasks')` using server actions during mutations, for ex [HERE](https://github.com/copilot-platforms/tasks-app/blob/fb44de20a05ad6833685408c12bccc623cb5b052/src/app/actions.ts#L22).
- After the first render, we can make use of SWR or similar to implement real-time updates using polling by fetching the data in intervals from the client side.